Source/WebCore/page/csp/ContentSecurityPolicy.cpp

```diff
@@ -848,7 +884,9 @@ void ContentSecurityPolicy::reportViolation(const String& effectiveViolatedDirec
     info.documentURI = m_documentURL ? m_documentURL.value().strippedForUseAsReferrer().string : blockedURI;
     info.lineNumber = sourcePosition.m_line.oneBasedInt();
     info.columnNumber = sourcePosition.m_column.oneBasedInt();
-    info.sample = violatedDirectiveList.shouldReportSample(effectiveViolatedDirective) ? sourceContent.left(40).toString() : emptyString();
+    info.sample = emptyString();
+    if (violatedDirectiveList.shouldReportSample(effectiveViolatedDirective))
+        info.sample = samplePreClipped ? sourceContent.toString() : sourceContent.left(40).toString();
```

Comment:
This is not necessary. If the source content is already short, then it’s harmless to do the `left(40)` on it. If you want to bump this number up from 40 to something a little higher I guess that’s OK. Please don’t add `samplePreClipped`. I guess the tests are maybe written expecting a certain very specific type of truncation, but seriously this is too messy.

Reply:
It's very specifically specced in this way currently.

I'm happy to tidy it up and just accept the test failures to keep this cleaner.